packages/origin.js/src/resources/purchases.js

```diff
@@ -1,48 +1,56 @@
-class Purchases {
-    constructor({ contractService, ipfsService }) {
-      this.contractService = contractService
-      this.ipfsService = ipfsService
-    }
+import ResourceBase from"../ResourceBase"
 
-    async contractFn(address, functionName, args=[], value=0){
-        const purchaseContract = this.contractService.purchaseContract
-        const purchase = await purchaseContract.at(address)
-        const account = await this.contractService.currentAccount()
-        args.push({ from: account, value: value })
-        return await purchase[functionName].apply(purchase, args)
-    }
 
-    async get(address){
-        const contractData = await this.contractFn(
-            address,
-            'data'
-        )
-        return {
-            address: address,
-            stage: contractData[0], // perhaps return a string
-            listingAddress: contractData[1],
-            buyerAddress: contractData[2],
-            created: contractData[3],
-            buyerTimout: contractData[4]
-        }
-    }
+class Purchases extends ResourceBase{
+  constructor({ contractService, ipfsService }) {
+    super({ contractService, ipfsService })
+    
+    this.contractDefinition = this.contractService.purchaseContract
 
-    async pay(address, amountWei){
-        return await this.contractFn(
-            address,
-            'pay',
-            [],
-            amountWei
-        )
+    this._STAGES_TO_NUMBER = {
+      awaiting_payment: 0,
+      shipping_pending: 1,
+      buyer_pending: 2,
+      seller_pending: 3,
+      in_dispute: 4,
+      review_period: 5,
+      complete: 6
```

Comment:
I don't think consumers of origin.js should have access to these stages as attributes. They should be able to see the string version of a particular stage value when interacting with origin.js, rather then the integer values, of course. But they should not have to access any properties on the purchase resource that just define the stages.

Because these are only for internal use, I'm pretty sure they should be defined as constants at the top of the file. Shouldn't be placed anywhere on the exported object.

(If we want to talk about sharing between other internal files, we can get fancier and define modules that simply contain constant definitions to share between files internally. But I don't think we need that yet.)